### ISSUE TITLE
Tensorflow-metal error when running .fit()

### ISSUE THREAD

**Issue by MichaelXCChen** [tensorflow/tensorflow#53441]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS 12.1
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.7.0
- Python version: 3.9.5
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: Apple Metal 32gb (unified)

The error occurs when I attempt to train the model using the GPU. The error does not happen when I specify `with tf.device('/cpu:0'):`, so CPU training appears to be fine, and returns similar results when cross-checked with the results obtained on Google Colab. This appears to be an issue specifically due to tensorflow-metal. My tensorflow-metal version is `0.3.0`. 

**Standalone code to reproduce the issue**
The example on [3D image classification from CT scans](https://keras.io/examples/vision/3D_image_classification/)
More specifically, the section that starts model training:

<pre><code># Compile model.
initial_learning_rate = 0.0001
lr_schedule = keras.optimizers.schedules.ExponentialDecay(
    initial_learning_rate, decay_steps=100000, decay_rate=0.96, staircase=True
)
model.compile(
    loss="binary_crossentropy",
    optimizer=keras.optimizers.Adam(learning_rate=lr_schedule),
    metrics=["acc"],
)

# Define callbacks.
checkpoint_cb = keras.callbacks.ModelCheckpoint(
    "3d_image_classification.h5", save_best_only=True
)
early_stopping_cb = keras.callbacks.EarlyStopping(monitor="val_acc", patience=15)

# Train the model, doing validation at the end of each epoch
epochs = 100

model.fit(
    train_dataset,
    validation_data=validation_dataset,
    epochs=epochs,
    shuffle=True,
    verbose=2,
    callbacks=[checkpoint_cb, early_stopping_cb],
)</code></pre>

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
[error.txt](https://github.com/tensorflow/tensorflow/files/7723232/error.txt)

**Comment by tilakrayal** [comment]
@MichaelXCChen ,
I was able to run the code without any issues on TF v2.7. Please find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/7d2e8ac14efb016f0552e904c1a749f7/3d_image_classification.ipynb).Could you please create a virtual environment and test your code again. It helps. Thanks!

**Comment by MichaelXCChen** [comment]
@tilakrayal 
Just wondering, did you use an conda environment using `conda create` or a python env `python3 -m venv`? I first installed Miniforge from [here](https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-MacOSX-arm64.sh), created a conda env, then installed:
`conda install -c apple tensorflow-deps==2.7.0`
`python3 -m pip install tensorflow-macos`
`python3 -m pip install tensorflow-metal`
`conda install jupyter pandas numpy matplotlib scikit-learn`

I have tried to create a fresh conda environment using the above process and the same error occurs. The thing is, the code works fine on Google Colab, but not locally on my M1 Max Macbook. The code also works fine when I force it to run on CPU, but the error occurs when I use the M1 GPU.

**Comment by tilakrayal** [comment]
@MichaelXCChen ,
Installation issues within the Anaconda environment are tracked in the Anaconda repo.As mentioned I was able to run the code without any issues on TF v2.7.Could you please create a virtual environment and test your code again. It helps. Thanks!

**Comment by MichaelXCChen** [comment]
@tilakrayal 
Thank you for the response. What version of tensorflow-metal are you running the code on? Is it 0.3.0?

**Comment by tilakrayal** [comment]
@MichaelXCChen ,
Since its not a bug from TF end and you also mentioned that the issue is with M1, Can you try posting your question in this tf-metal [developer forum of Apple](https://developer.apple.com/forums/tags/tensorflow-metal)? Thanks!
